app/code/Magento/ConfigurableProduct/Helper/Data.php

```diff
@@ -90,7 +98,8 @@ public function getOptions($currentProduct, $allowedProducts)
                 $productAttribute = $attribute->getProductAttribute();
                 $productAttributeId = $productAttribute->getId();
                 $attributeValue = $product->getData($productAttribute->getAttributeCode());
-                if ($product->isSalable()) {
+                $salableqty = $this->getSalableQuantityDataBySku->execute($product->getSku());
```

Comment:
This implementation is not fully correct:
 - what if stack management is disabled
 - what if back orders is enabled
 
 Also , ConfigurableProduct must not depend on InventorySalesAdminUI module